### ISSUE TITLE
add option to register several same middlewares (for different routes)

### ISSUE THREAD

**Issue by Baroshem** [Baroshem/nuxt-security#38]

**Comment by Baroshem** [comment]
Allow to pass an array of routes or single route so that multiple middlewares could be registered with the same value

Or

Allow to pass multiple configuration objects so that user can configure the same middleware multiple times but already with supported multiple values

**Comment by Baroshem** [comment]
This feature has proven to be much more difficult to implement that I expected. 

The first issue I encountered is the fact that defu is merging the default config with the one provided by the user. But it also flattens the object provided by the user so when user passes an array of objects to certain middleware config, it will be transformed to object with indexes as keys so defu cannot merge it.

The other solution would be to create array for both value and route of each middleware and then map 1:1 them in the middlewares but I have encountered issues with this solution as well.

For now, I will release 0.8.0 release with Nuxt 3 stable support and BasicAuth and in the 0.9.0 I will focus primarly on releasing support for this.

I think the solution here would be to map the newly created config by defu and if it contains indexes as keys (array was passed) then strip the default config out of it.
